core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.spec.ts

```diff
@@ -141,6 +167,313 @@ describe('ExplorationFooterComponent', () => {
     expect(component.hintsAndSolutionsAreSupported).toBeFalse();
   });
 
+  it('should push 1 in checkpointArray when index < 1', () => {
+    let ngbModal = TestBed.inject(NgbModal);
+
+    expect(component.checkpointArray).toEqual([0]);
+
+    spyOn(playerPositionService, 'getDisplayedCardIndex').and.
+      returnValue(0);
+
+    spyOn(ngbModal, 'open').and.returnValue({
+      componentInstance: {
+        numberofCheckpoints: 0,
+        completedWidth: 0,
+        contributorNames: [],
+        expInfo: null
+      },
+      result: {
+        then: (successCallback: () => void, errorCallback: () => void) => {
+          successCallback();
+          errorCallback();
+        }
+      }
+    } as NgbModalRef);
+
+    component.openInformationCardModal();
+
+    expect(ngbModal.open).toHaveBeenCalled();
+    expect(component.checkpointArray).toEqual([0, 1]);
+  });
+
+  it('should push 0 in checkpointArray when card_is_checkpoint' +
```

Comment:
So, the two tests address if-else conditions. If `card_is_checkpoint` is true, 1 is pushed else 0 is pushed. In my opinion, combining them would lead to reduced clarity. What do you think? I'll surely update the test names to be more meaningful.

Reply:
looks better now